src/gstools/field/srf.py

```diff
@@ -142,6 +149,11 @@ def __call__(
         field : :class:`numpy.ndarray`
             the SRF
         """
+        if isinstance(self.generator, Fourier) and mesh_type != "structured":
+            raise ValueError(
+                "SRF: Fourier generator only defined for "
+                'mesh_type == "structured".'
+            )
```

Comment:
Does it really need to be structured? I think it is passed unstructured to the generator either way and since the period is not determined from the pos tuple anymore, this check could be dropped.

Reply:
For the Fourier method to work, the spatial grid has to be regular, so this is a convenient way to ensure that. Otherwise, I could imagine that users might run into quite obscure problems with the field generation. Maybe we could add that feature, when it is really needed?

Reply:
But the generator gets `iso_pos` which are always unstructured and from going through the summator I don't see where it matters that the original pos tuple was defining a structured grid. Isn't the whole point about the Fourier generator, that it is periodic? And since the periods in each main direction are now given independently from the field, I think this should just work.

Imagine the follwing scenario: We have a 2x2 grid. If we only pass two points from that (top-left, bottom-right) as an unstructured grid to the summator I should get the same results for these two points like within the structured 2x2 grid. Since the summator calulates the results for each point independently. Scaling this example, for an arbitrary unstructured grid you could generate a structured grid containing all points from the unstructured one. Now generate the field and cut out the unstructured grid. This should again be the same field as passing the original unstructured field. So I still think it is fine to use unstructured grids.

Or are the other restrictions? Like that the fields needs to be wihtin the first period or something similar?

Reply:
You are completely right that the Fourier method works with a regular (Euclidean) mesh passed to GSTools as what we call an unstructured mesh. However, the Fourier method is only defined on regular meshes. By throwing that exception, we can easily prevent problems, which will be hard to catch later, as they are pretty subtle. Of course, with a structured mesh, you can still create meshes, with non-equidistant steps, but I guess that happens very seldomly.
Regarding your last sentence: No, the periodicity can be larger and smaller than the domain size.
But in general, the periodicity breaks on unstructured meshes, which makes the keyword `period` a bit strange. Here is the output of a slightly adapted version of your tutorial with the Fourier method. The periodicity should be about half the domain size.

![unstruct_fourier](https://github.com/GeoStat-Framework/GSTools/assets/26794562/fa2b7bae-319e-4fa8-9bea-b19f99e3454a)

I don't know, I could drop the exception and print a warning, about the features of the unstructured mesh or put a very clear warning in the docs?

Reply:
By the way, at some point we should find an alternative to meshzoo ;-)

Reply:
Okay, after some testing, the Fourier method *does* indeed work on non-equidistant meshes, so dropping that exception.